### ISSUE TITLE
Lots of test errors with Python 3.14.0a2

### ISSUE THREAD

**Issue by hroncok** [kislyuk/argcomplete#515]
When we build argcomplete in Fedora with Python 3.14, we get a lot of failures.

This can be reproduced e.g. with:

```
$ uv venv --python=python3.14 venv --seed
$ . venv/bin/activate
$ python --version
Python 3.14.0a2
$ make install
...
Successfully installed argcomplete-3.5.1 coverage-7.6.8 mypy-1.13.0 mypy-extensions-1.0.0 pexpect-4.9.0 ptyprocess-0.7.0 ruff-0.8.0 typing-extensions-4.12.2 wheel-0.45.1
$ make test
coverage run --source=argcomplete --omit=argcomplete/packages/_shlex.py ./test/test.py -v
test_action_activation (__main__.TestArgcomplete.test_action_activation) ... ERROR
test_action_activation_with_subparser (__main__.TestArgcomplete.test_action_activation_with_subparser) ... ERROR
test_append_space (__main__.TestArgcomplete.test_append_space) ... ERROR
test_basic_completion (__main__.TestArgcomplete.test_basic_completion) ... ERROR
test_choices (__main__.TestArgcomplete.test_choices) ... ERROR
test_completers (__main__.TestArgcomplete.test_completers) ... ERROR
test_custom_validator (__main__.TestArgcomplete.test_custom_validator) ... ERROR
test_default_completer (__main__.TestArgcomplete.test_default_completer) ... ERROR
test_different_validators (__main__.TestArgcomplete.test_different_validators) ... ERROR
test_directory_completion (__main__.TestArgcomplete.test_directory_completion) ... ok
test_display_completions (__main__.TestArgcomplete.test_display_completions) ... ERROR
test_display_completions_with_aliases (__main__.TestArgcomplete.test_display_completions_with_aliases) ... ERROR
test_escape_special_chars (__main__.TestArgcomplete.test_escape_special_chars) ... ERROR
test_exclusive (__main__.TestArgcomplete.test_exclusive) ... ERROR
test_exclusive_class (__main__.TestArgcomplete.test_exclusive_class) ... ERROR
test_file_completion (__main__.TestArgcomplete.test_file_completion) ... ok
test_filescompleter_filetype_integration (__main__.TestArgcomplete.test_filescompleter_filetype_integration) ... /home/churchyard/Dokumenty/RedHat/argcomplete/test/test.py:362: PendingDeprecationWarning: FileType is deprecated. Simply open files after parsing arguments.
  parser.add_argument("--r", type=argparse.FileType("r"))
/home/churchyard/Dokumenty/RedHat/argcomplete/test/test.py:363: PendingDeprecationWarning: FileType is deprecated. Simply open files after parsing arguments.
  parser.add_argument("--w", type=argparse.FileType("w"))
ERROR
test_mixed_optional_positional (__main__.TestArgcomplete.test_mixed_optional_positional) ... ERROR
test_nargs_one_or_more (__main__.TestArgcomplete.test_nargs_one_or_more) ... ERROR
test_nargs_optional (__main__.TestArgcomplete.test_nargs_optional) ... ERROR
test_nargs_zero_or_more (__main__.TestArgcomplete.test_nargs_zero_or_more) ... ERROR
test_non_ascii (__main__.TestArgcomplete.test_non_ascii) ... ERROR
test_non_str_choices (__main__.TestArgcomplete.test_non_str_choices) ... ERROR
test_option_help (__main__.TestArgcomplete.test_option_help) ... ERROR
test_optional_long_short_filtering (__main__.TestArgcomplete.test_optional_long_short_filtering) ... ERROR
test_optional_nargs (__main__.TestArgcomplete.test_optional_nargs) ... ERROR
test_positional_remainder (__main__.TestArgcomplete.test_positional_remainder) ... ERROR
test_readline_entry_point (__main__.TestArgcomplete.test_readline_entry_point) ... ERROR
test_shellcode_utility (__main__.TestArgcomplete.test_shellcode_utility) ... ok
test_skipped_completer (__main__.TestArgcomplete.test_skipped_completer) ... ERROR
test_subparser_completers (__main__.TestArgcomplete.test_subparser_completers) ... ERROR
test_subparsers (__main__.TestArgcomplete.test_subparsers) ... ERROR
test_suppress_args (__main__.TestArgcomplete.test_suppress_args) ... ERROR
test_suppress_completer (__main__.TestArgcomplete.test_suppress_completer) ... ERROR
test_repl_multiple_complete (__main__.TestArgcompleteREPL.test_repl_multiple_complete) ... ERROR
test_repl_parse_after_complete (__main__.TestArgcompleteREPL.test_repl_parse_after_complete) ... ERROR
test_repl_reuse_parser_with_positional (__main__.TestArgcompleteREPL.test_repl_reuse_parser_with_positional) ... ERROR
test_repl_subcommand (__main__.TestArgcompleteREPL.test_repl_subcommand) ... ERROR
test_repl_subparser_parse_after_complete (__main__.TestArgcompleteREPL.test_repl_subparser_parse_after_complete) ... ERROR
test_comp_point (__main__.TestBash.test_comp_point) ... FAIL
test_completion_environment (__main__.TestBash.test_completion_environment) ... FAIL
test_continuation (__main__.TestBash.test_continuation) ... FAIL
test_debug_output (__main__.TestBash.test_debug_output) ... FAIL
test_double_quoted_completion (__main__.TestBash.test_double_quoted_completion) ... ERROR
test_exclamation_in_double_quotes (__main__.TestBash.test_exclamation_in_double_quotes) ... expected failure
test_nounset (__main__.TestBash.test_nounset) ... FAIL
test_one_space_after_exact (__main__.TestBash.test_one_space_after_exact)
Test exactly one space is appended after an exact match. ... FAIL
test_parse_special_characters (__main__.TestBash.test_parse_special_characters) ... FAIL
test_parse_special_characters_dollar (__main__.TestBash.test_parse_special_characters_dollar) ... expected failure
test_partial_completion (__main__.TestBash.test_partial_completion) ... FAIL
test_quoted_exact (__main__.TestBash.test_quoted_exact) ... ERROR
test_quoted_space (__main__.TestBash.test_quoted_space) ... ERROR
test_quotes (__main__.TestBash.test_quotes) ... FAIL
test_simple_completion (__main__.TestBash.test_simple_completion) ... FAIL
test_single_quoted_completion (__main__.TestBash.test_single_quoted_completion) ... ERROR
test_single_quotes_in_double_quotes (__main__.TestBash.test_single_quotes_in_double_quotes) ... ERROR
test_single_quotes_in_single_quotes (__main__.TestBash.test_single_quotes_in_single_quotes) ... ERROR
test_special_characters (__main__.TestBash.test_special_characters) ... FAIL
test_special_characters_double_quoted (__main__.TestBash.test_special_characters_double_quoted) ... FAIL
test_special_characters_single_quoted (__main__.TestBash.test_special_characters_single_quoted) ... FAIL
test_temp_file (__main__.TestBash.test_temp_file) ... FAIL
test_unquoted_space (__main__.TestBash.test_unquoted_space) ... FAIL
test_wordbreak_chars (__main__.TestBash.test_wordbreak_chars) ... FAIL
test_comp_point (__main__.TestBashGlobal.test_comp_point) ... FAIL
test_completion_environment (__main__.TestBashGlobal.test_completion_environment) ... FAIL
test_console_script_module (__main__.TestBashGlobal.test_console_script_module)
Test completing a console_script for a module. ... FAIL
test_console_script_module_wheel (__main__.TestBashGlobal.test_console_script_module_wheel)
Test completing a console_script for a module from a wheel. ... FAIL
test_console_script_package (__main__.TestBashGlobal.test_console_script_package)
Test completing a console_script for a package. ... FAIL
test_console_script_package_wheel (__main__.TestBashGlobal.test_console_script_package_wheel)
Test completing a console_script for a package from a wheel. ... FAIL
test_continuation (__main__.TestBashGlobal.test_continuation) ... FAIL
test_debug_output (__main__.TestBashGlobal.test_debug_output) ... FAIL
test_double_quoted_completion (__main__.TestBashGlobal.test_double_quoted_completion) ... ERROR
test_exclamation_in_double_quotes (__main__.TestBashGlobal.test_exclamation_in_double_quotes) ... expected failure
test_nounset (__main__.TestBashGlobal.test_nounset) ... FAIL
test_one_space_after_exact (__main__.TestBashGlobal.test_one_space_after_exact)
Test exactly one space is appended after an exact match. ... FAIL
test_parse_special_characters (__main__.TestBashGlobal.test_parse_special_characters) ... FAIL
test_parse_special_characters_dollar (__main__.TestBashGlobal.test_parse_special_characters_dollar) ... expected failure
test_partial_completion (__main__.TestBashGlobal.test_partial_completion) ... FAIL
test_python_completion (__main__.TestBashGlobal.test_python_completion) ... FAIL
test_python_filename_completion (__main__.TestBashGlobal.test_python_filename_completion) ... FAIL
test_python_module (__main__.TestBashGlobal.test_python_module)
Test completing a module run with python -m. ... FAIL
test_python_not_executable (__main__.TestBashGlobal.test_python_not_executable)
Test completing a script that cannot be run directly. ... FAIL
test_python_stuck (__main__.TestBashGlobal.test_python_stuck) ... ok
test_quoted_exact (__main__.TestBashGlobal.test_quoted_exact) ... ERROR
test_quoted_space (__main__.TestBashGlobal.test_quoted_space) ... ERROR
test_quotes (__main__.TestBashGlobal.test_quotes) ... FAIL
test_redirection_completion (__main__.TestBashGlobal.test_redirection_completion) ... ok
test_simple_completion (__main__.TestBashGlobal.test_simple_completion) ... FAIL
test_single_quoted_completion (__main__.TestBashGlobal.test_single_quoted_completion) ... ERROR
test_single_quotes_in_double_quotes (__main__.TestBashGlobal.test_single_quotes_in_double_quotes) ... ERROR
test_single_quotes_in_single_quotes (__main__.TestBashGlobal.test_single_quotes_in_single_quotes) ... ERROR
test_special_characters (__main__.TestBashGlobal.test_special_characters) ... FAIL
test_special_characters_double_quoted (__main__.TestBashGlobal.test_special_characters_double_quoted) ... FAIL
test_special_characters_single_quoted (__main__.TestBashGlobal.test_special_characters_single_quoted) ... FAIL
test_temp_file (__main__.TestBashGlobal.test_temp_file) ... FAIL
test_unquoted_space (__main__.TestBashGlobal.test_unquoted_space) ... FAIL
test_wordbreak_chars (__main__.TestBashGlobal.test_wordbreak_chars) ... FAIL
test_module (__main__.TestCheckModule.test_module) ... ok
test_not_package (__main__.TestCheckModule.test_not_package) ... ok
test_package (__main__.TestCheckModule.test_package) ... ok
test_package_main (__main__.TestCheckModule.test_package_main) ... ok
test_subpackage (__main__.TestCheckModule.test_subpackage) ... ok
test_escaped_special (__main__.TestSplitLine.test_escaped_special) ... ok
test_escaped_special_in_double_quotes (__main__.TestSplitLine.test_escaped_special_in_double_quotes) ... expected failure
test_last_wordbreak_pos (__main__.TestSplitLine.test_last_wordbreak_pos) ... ok
test_punctuation (__main__.TestSplitLine.test_punctuation) ... ok
test_simple (__main__.TestSplitLine.test_simple) ... ok
test_unescaped_special (__main__.TestSplitLine.test_unescaped_special) ... ok
test_comp_point (__main__.TestZsh.test_comp_point) ... skipped 'skip'
test_completion_environment (__main__.TestZsh.test_completion_environment) ... skipped 'skip'
test_continuation (__main__.TestZsh.test_continuation) ... FAIL
test_double_quoted_completion (__main__.TestZsh.test_double_quoted_completion) ... ERROR
test_exclamation_in_double_quotes (__main__.TestZsh.test_exclamation_in_double_quotes) ... ERROR
test_parse_special_characters (__main__.TestZsh.test_parse_special_characters) ... skipped 'skip'
test_parse_special_characters_dollar (__main__.TestZsh.test_parse_special_characters_dollar) ... skipped 'skip'
test_partial_completion (__main__.TestZsh.test_partial_completion) ... FAIL
test_quoted_exact (__main__.TestZsh.test_quoted_exact) ... ERROR
test_quoted_space (__main__.TestZsh.test_quoted_space) ... ERROR
test_quotes (__main__.TestZsh.test_quotes) ... FAIL
test_simple_completion (__main__.TestZsh.test_simple_completion) ... FAIL
test_single_quoted_completion (__main__.TestZsh.test_single_quoted_completion) ... ERROR
test_single_quotes_in_double_quotes (__main__.TestZsh.test_single_quotes_in_double_quotes) ... ERROR
test_single_quotes_in_single_quotes (__main__.TestZsh.test_single_quotes_in_single_quotes) ... ERROR
test_special_characters (__main__.TestZsh.test_special_characters) ... FAIL
test_special_characters_double_quoted (__main__.TestZsh.test_special_characters_double_quoted) ... FAIL
test_special_characters_single_quoted (__main__.TestZsh.test_special_characters_single_quoted) ... FAIL
test_unquoted_space (__main__.TestZsh.test_unquoted_space) ... FAIL
test_wordbreak_chars (__main__.TestZsh.test_wordbreak_chars) ... FAIL
test_comp_point (__main__.TestZshGlobalExplicit.test_comp_point) ... skipped 'skip'
test_completion_environment (__main__.TestZshGlobalExplicit.test_completion_environment) ... skipped 'skip'
test_console_script_module (__main__.TestZshGlobalExplicit.test_console_script_module)
Test completing a console_script for a module. ... FAIL
test_console_script_module_wheel (__main__.TestZshGlobalExplicit.test_console_script_module_wheel)
Test completing a console_script for a module from a wheel. ... FAIL
test_console_script_package (__main__.TestZshGlobalExplicit.test_console_script_package)
Test completing a console_script for a package. ... FAIL
test_console_script_package_wheel (__main__.TestZshGlobalExplicit.test_console_script_package_wheel)
Test completing a console_script for a package from a wheel. ... FAIL
test_continuation (__main__.TestZshGlobalExplicit.test_continuation) ... FAIL
test_double_quoted_completion (__main__.TestZshGlobalExplicit.test_double_quoted_completion) ... ERROR
test_exclamation_in_double_quotes (__main__.TestZshGlobalExplicit.test_exclamation_in_double_quotes) ... ERROR
test_parse_special_characters (__main__.TestZshGlobalExplicit.test_parse_special_characters) ... skipped 'skip'
test_parse_special_characters_dollar (__main__.TestZshGlobalExplicit.test_parse_special_characters_dollar) ... skipped 'skip'
test_partial_completion (__main__.TestZshGlobalExplicit.test_partial_completion) ... FAIL
test_python_completion (__main__.TestZshGlobalExplicit.test_python_completion) ... FAIL
test_python_filename_completion (__main__.TestZshGlobalExplicit.test_python_filename_completion) ... FAIL
test_python_module (__main__.TestZshGlobalExplicit.test_python_module)
Test completing a module run with python -m. ... FAIL
test_python_not_executable (__main__.TestZshGlobalExplicit.test_python_not_executable)
Test completing a script that cannot be run directly. ... FAIL
test_python_stuck (__main__.TestZshGlobalExplicit.test_python_stuck) ... ok
test_quoted_exact (__main__.TestZshGlobalExplicit.test_quoted_exact) ... ERROR
test_quoted_space (__main__.TestZshGlobalExplicit.test_quoted_space) ... ERROR
test_quotes (__main__.TestZshGlobalExplicit.test_quotes) ... FAIL
test_redirection_completion (__main__.TestZshGlobalExplicit.test_redirection_completion) ... ok
test_simple_completion (__main__.TestZshGlobalExplicit.test_simple_completion) ... FAIL
test_single_quoted_completion (__main__.TestZshGlobalExplicit.test_single_quoted_completion) ... ERROR
test_single_quotes_in_double_quotes (__main__.TestZshGlobalExplicit.test_single_quotes_in_double_quotes) ... ERROR
test_single_quotes_in_single_quotes (__main__.TestZshGlobalExplicit.test_single_quotes_in_single_quotes) ... ERROR
test_special_characters (__main__.TestZshGlobalExplicit.test_special_characters) ... FAIL
test_special_characters_double_quoted (__main__.TestZshGlobalExplicit.test_special_characters_double_quoted) ... FAIL
test_special_characters_single_quoted (__main__.TestZshGlobalExplicit.test_special_characters_single_quoted) ... FAIL
test_unquoted_space (__main__.TestZshGlobalExplicit.test_unquoted_space) ... FAIL
test_wordbreak_chars (__main__.TestZshGlobalExplicit.test_wordbreak_chars) ... FAIL
test_comp_point (__main__.TestZshGlobalImplicit.test_comp_point) ... skipped 'skip'
test_completion_environment (__main__.TestZshGlobalImplicit.test_completion_environment) ... skipped 'skip'
test_console_script_module (__main__.TestZshGlobalImplicit.test_console_script_module)
Test completing a console_script for a module. ... FAIL
test_console_script_module_wheel (__main__.TestZshGlobalImplicit.test_console_script_module_wheel)
Test completing a console_script for a module from a wheel. ... FAIL
test_console_script_package (__main__.TestZshGlobalImplicit.test_console_script_package)
Test completing a console_script for a package. ... FAIL
test_console_script_package_wheel (__main__.TestZshGlobalImplicit.test_console_script_package_wheel)
Test completing a console_script for a package from a wheel. ... FAIL
test_continuation (__main__.TestZshGlobalImplicit.test_continuation) ... FAIL
test_double_quoted_completion (__main__.TestZshGlobalImplicit.test_double_quoted_completion) ... ERROR
test_exclamation_in_double_quotes (__main__.TestZshGlobalImplicit.test_exclamation_in_double_quotes) ... ERROR
test_parse_special_characters (__main__.TestZshGlobalImplicit.test_parse_special_characters) ... skipped 'skip'
test_parse_special_characters_dollar (__main__.TestZshGlobalImplicit.test_parse_special_characters_dollar) ... skipped 'skip'
test_partial_completion (__main__.TestZshGlobalImplicit.test_partial_completion) ... FAIL
test_python_completion (__main__.TestZshGlobalImplicit.test_python_completion) ... FAIL
test_python_filename_completion (__main__.TestZshGlobalImplicit.test_python_filename_completion) ... FAIL
test_python_module (__main__.TestZshGlobalImplicit.test_python_module)
Test completing a module run with python -m. ... FAIL
test_python_not_executable (__main__.TestZshGlobalImplicit.test_python_not_executable)
Test completing a script that cannot be run directly. ... FAIL
test_python_stuck (__main__.TestZshGlobalImplicit.test_python_stuck) ... ok
test_quoted_exact (__main__.TestZshGlobalImplicit.test_quoted_exact) ... ERROR
test_quoted_space (__main__.TestZshGlobalImplicit.test_quoted_space) ... ERROR
test_quotes (__main__.TestZshGlobalImplicit.test_quotes) ... FAIL
test_redirection_completion (__main__.TestZshGlobalImplicit.test_redirection_completion) ... ok
test_simple_completion (__main__.TestZshGlobalImplicit.test_simple_completion) ... FAIL
test_single_quoted_completion (__main__.TestZshGlobalImplicit.test_single_quoted_completion) ... ERROR
test_single_quotes_in_double_quotes (__main__.TestZshGlobalImplicit.test_single_quotes_in_double_quotes) ... ERROR
test_single_quotes_in_single_quotes (__main__.TestZshGlobalImplicit.test_single_quotes_in_single_quotes) ... ERROR
test_special_characters (__main__.TestZshGlobalImplicit.test_special_characters) ... FAIL
test_special_characters_double_quoted (__main__.TestZshGlobalImplicit.test_special_characters_double_quoted) ... FAIL
test_special_characters_single_quoted (__main__.TestZshGlobalImplicit.test_special_characters_single_quoted) ... FAIL
test_unquoted_space (__main__.TestZshGlobalImplicit.test_unquoted_space) ... FAIL
test_wordbreak_chars (__main__.TestZshGlobalImplicit.test_wordbreak_chars) ... FAIL
test_warn (__main__.Warn.test_warn) ... ok
...
FAILED (failures=83, errors=69, skipped=12, expected failures=5)
```

The full log is too old for the issues, so it is attached: [test-failures.txt](https://github.com/user-attachments/files/17903448/test-failures.txt)

The errors/failures look like:

```
ERROR: test_append_space (__main__.TestArgcomplete.test_append_space)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../argcomplete/test/test.py", line 802, in test_append_space
    self.assertEqual(self.run_completer(make_parser(), "prog "), ["bar "])
                     ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../argcomplete/test/test.py", line 139, in run_completer
    completer(parser, output_stream=t, exit_method=sys.exit, **kwargs)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../argcomplete/argcomplete/finders.py", line 174, in __call__
    completions = self._get_completions(comp_words, cword_prefix, cword_prequote, last_wordbreak_pos)
  File ".../argcomplete/argcomplete/finders.py", line 223, in _get_completions
    completions = self.collect_completions(active_parsers, parsed_args, cword_prefix)
  File ".../argcomplete/argcomplete/finders.py", line 440, in collect_completions
    completions += self._get_option_completions(active_parser, cword_prefix)
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../argcomplete/argcomplete/finders.py", line 326, in _get_option_completions
    if not self._action_allowed(action, parser):
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File ".../argcomplete/argcomplete/finders.py", line 336, in _action_allowed
    for conflict_action in parser._action_conflicts.get(action, []):
                           ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MonkeyPatchedIntrospectiveArgumentParser' object has no attribute '_action_conflicts'

======================================================================
ERROR: test_quoted_exact (__main__.TestBash.test_quoted_exact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../argcomplete/test/test.py", line 1191, in test_quoted_exact
    self.assertEqual(self.sh.run_command('prog basic "f\t--'), "foo\r\n")
                     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File ".../argcomplete/test/test.py", line 51, in run_command
    res = super().run_command(command, **kwargs)
  File ".../argcomplete/venv/lib64/python3.14/site-packages/pexpect/replwrap.py", line 107, in run_command
    raise ValueError("Continuation prompt found - input was incomplete:\n"
                     + command)
ValueError: Continuation prompt found - input was incomplete:
prog basic "f	--

======================================================================
FAIL: test_simple_completion (__main__.TestBash.test_simple_completion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../argcomplete/test/test.py", line 1165, in test_simple_completion
    self.assertEqual(self.sh.run_command("prog basic f\t"), "foo\r\n")
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: "usage: prog basic [-h] {foo,bar,baz}\r\nprog basic: error: argument arg: invalid choice: 'f' (choose from foo, bar, baz)\r\n" != 'foo\r\n'
+ foo
- usage: prog basic [-h] {foo,bar,baz}
- prog basic: error: argument arg: invalid choice: 'f' (choose from foo, bar, baz)
```

**Comment by kislyuk** [comment]
This should be fixed in #513.

**Comment by kislyuk** [comment]
Never mind, #513 is about Python 3.13.

**Comment by kislyuk** [comment]
Thanks for reporting - let's hold off on this until Python 3.14 is closer to release.

**Comment by kislyuk** [comment]
OK, the relevant change made it in to Python 3.12 and 3.13. Released argcomplete v3.5.2 with a fix to restore compatibility.
